File: aat-android/src/main/java/ch/bailu/aat/services/elevation/tile/Dem3Lock.java
package ch.bailu.aat.services.elevation.tile;

import ch.bailu.aat_lib.logger.AppLog;

public final class Dem3Lock {

    int locks=0;

    public void lock(Object owner) {
        locks++;

    }


    public void free(Object owner) {
        locks--;
    }


    public boolean isLocked() {
        if (locks < 0) AppLog.w(this, "Negative lock!!: " + locks);
        return locks>0;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/elevation/Dem3Lock.kt
package ch.bailu.aat_lib.service.elevation

import ch.bailu.aat_lib.logger.AppLog.w

class Dem3Lock {
    private var locks: Int = 0

    fun lock() {
        locks++
    }

    fun free() {
        locks--
    }

    val isLocked: Boolean
        get() {
            if (locks < 0) {
                w(this, "Negative lock!!: $locks")
            }
            return locks > 0
        }
}
